src/Refactoring-Core/RBRenameClassVariableRefactoring.class.st

```diff
@@ -34,6 +34,21 @@ RBRenameClassVariableRefactoring class >> rename: aVarName to: aName in: aClass
 		in: aClass
 ]
 
+{ #category : 'preconditions' }
+RBRenameClassVariableRefactoring >> applicabilityPreconditions [
+
+	^ {
+		  (RBIsNotAMetaclass class: class).
+		  (RBIsValidSharedVariableName name: newName).
+		  (RBIsVariableNotDefinedInHierarchy name: newName class: class).
+		  (RBNameIsGlobalCondition new model: model className: newName) not.
+		  "For now only rename a shared variable that is locally defined in a class."
```

Comment:
@Ducasse @balsa-sarenac do you agree on adding this condition here and in the RenameInstanceVariableRefactoring?

Reply:
Ok I checked the existing code and yes this is really an applicabilityPreconditions